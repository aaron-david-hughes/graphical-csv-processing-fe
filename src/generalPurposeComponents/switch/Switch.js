import React from 'react';
import InternalSwitch from "react-switch";

class Switch extends React.Component {
    render() {
        return <div id={this.props.title + '-switch'}>
            <label>
                <InternalSwitch
                    checked={this.props.isChecked}
                    onChange={this.props.onChange}
                />
                {this.props.title}
            </label>
        </div>
    }
}

export default Switch;