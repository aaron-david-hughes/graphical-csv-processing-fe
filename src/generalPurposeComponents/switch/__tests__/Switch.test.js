import React from 'react';
import {shallow} from 'enzyme';
import Switch from '../Switch';

describe('Switch component tests', () => {
    let render;

    beforeEach(() => {
        render = shallow(<Switch title='switchTestTitle' isChecked={true} onChange={jest.fn()}/>)
    });

    describe('render tests', () => {
        it('should match snapshot', () => {
            expect(render).toMatchSnapshot();
        });

        it('should have title matching props', () => {
            let title = render.find('label').text();

            expect(title).toContain('switchTestTitle');
        });
    });
});