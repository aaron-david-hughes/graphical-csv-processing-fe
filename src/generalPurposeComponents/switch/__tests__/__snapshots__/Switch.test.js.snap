// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch component tests render tests should match snapshot 1`] = `
<div
  id="switchTestTitle-switch"
>
  <label>
    <ReactSwitch
      activeBoxShadow="0 0 2px 3px #3bf"
      boxShadow={null}
      checked={true}
      checkedIcon={
        <svg
          height="100%"
          style={
            Object {
              "position": "absolute",
              "top": 0,
            }
          }
          viewBox="-2 -5 17 21"
          width="100%"
        >
          <path
            d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
            fill="#fff"
            fillRule="evenodd"
          />
        </svg>
      }
      disabled={false}
      height={28}
      offColor="#888"
      offHandleColor="#fff"
      onChange={[MockFunction]}
      onColor="#080"
      onHandleColor="#fff"
      uncheckedIcon={
        <svg
          height="100%"
          style={
            Object {
              "position": "absolute",
              "top": 0,
            }
          }
          viewBox="-2 -5 14 20"
          width="100%"
        >
          <path
            d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
            fill="#fff"
            fillRule="evenodd"
          />
        </svg>
      }
      width={56}
    />
    switchTestTitle
  </label>
</div>
`;
